watertrading/code/BuildModel.py

```diff
@@ -0,0 +1,1361 @@
+    )
+    print("Inbound arc set complete")
+    
+    # set for arcs outbound from (pp,t)
+    def s_LM_out_pt_INIT(model,p_prime,t_prime):
+        elems = []
+        for (mi,p,c,tp,tc) in model.s_LLM:
+            if p == p_prime and tp == t_prime:
+                elems.append((mi,p,c,tp,tc))
+        # elems = [(mi,p,c,tp,tc) in model.s_LLM if p == p_prime and tp == t_prime]
+        return elems
+    model.s_LM_out_pt = Set(
+        model.s_PP,
+        model.s_T,
+        dimen=5,
+        initialize=s_LM_out_pt_INIT,
+        doc="Outbound arcs from production pad pp at time t"
+    )
+    print("Outbound arc set complete")
+
+    # set for all outbound arcs at time tp
+    def s_LM_out_mit_INIT(model, mi_prime, t_prime):
+        elems = []
+        for (mi,p,c,tp,tc) in model.s_LLM:
+            if mi == mi_prime and tp == t_prime:
+                elems.append((mi,p,c,tp,tc))
+        return elems
+    model.s_LM_out_mit = Set(
+        model.s_MI,
+        model.s_T,
+        within=model.s_LLM,
+        initialize=s_LM_out_mit_INIT,
+        doc="Outbound arcs from all pads pp at t for mi",
+    )
+
+    # All sets done
+    print("All sets complete")
+
+    ### ---------------------- VARIABLES & BOUNDS ---------------------- #
+    model.v_FP_Trucked = Var(
+        model.s_LLT,
+        within=NonNegativeReals,
+        initialize=0,
+        units=model.model_units["volume_time"],
+        doc="Produced water quantity trucked from location l to location l' [volume/time] by producer pi",
+    )
+
+    model.v_FM_Trucked = Var(
+        model.s_LLM,
+        within=NonNegativeReals,
+        initialize=0,
+        units=model.model_units["volume_time"],
+        doc="Produced water quantity trucked from location l to location l' [volume/time] by midstream mi",
+    )
+
+    model.v_Supply = Var(
+        model.s_PI,
+        model.s_T,
+        within=NonNegativeReals,
+        initialize=0,
+        units=model.model_units["volume_time"],
+        doc="Produced water volume supplied by producer request index pi at time t"
+    )
+
+    model.v_Demand = Var(
+        model.s_CI,
+        model.s_T,
+        within=NonNegativeReals,
+        initialize=0,
+        units=model.model_units["volume_time"],
+        doc="Produced water volume consumed by consumer request index ci at time t"
+    )
+    print("All variables set up")
+    
+    ### Variable Bounds
+    """
+    # v_FP_Trucked upper bound
+    NodalSupply = dict(((p,t),0.0) for p in model.s_PP for t in model.s_T)
+    for pi in model.s_PI:
+        for t in model.s_T_pi[pi]:
+            NodalSupply[Producer_Wellpad[pi],t] += Producer_Rate[pi]
+    
+    NodalDemand = dict(((c,t),0.0) for c in model.s_CP for t in model.s_T)
+    for ci in model.s_CI:
+        for t in model.s_T_ci[ci]:
+            NodalDemand[Consumer_Wellpad[ci],t] += Consumer_Rate[ci]
+    
+    def p_FP_Trucked_UB_init(model, p, c, t):
+        return min(NodalSupply[p,t], NodalDemand[c,t])
+    """
```

Comment:
Similarly recommend removing the above block of code.

Reply:
addressed